<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="$(env ROBOT_TYPE)" doc="Robot type: [a1, aliengo, go1, laikago]"/>

    <arg name="cheater" default="false"/>

    <arg name="urdfFile" default="$(find legged_unitree_description)/urdf/$(arg robot_type).urdf"/>
    <arg name="taskFile" default="$(find legged_controllers)/config/task/$(arg robot_type).info"/>
    <arg name="referenceFile" default="$(find legged_controllers)/config/reference/$(arg robot_type).info"/>
    <arg name="gaitCommandFile" default="$(find legged_controllers)/config/gait.info"/>

    <!-- make the files into global parameters -->
    <param name="urdfFile" value="$(arg urdfFile)"/>
    <param name="taskFile" value="$(arg taskFile)"/>
    <param name="referenceFile" value="$(arg referenceFile)"/>
    <param name="gaitCommandFile" value="$(arg gaitCommandFile)"/>

    <rosparam file="$(find legged_controllers)/config/controllers.yaml" command="load"/>

    <node if="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller
          controllers/legged_cheater_controller
"/>

    <node unless="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller
          controllers/legged_cmpc_controller
"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen"/>

    <node pkg="legged_controllers" type="legged_target_trajectories_publisher" name="legged_robot_target"
          output="screen"/>
</launch>
